Added hex-blob option for mysql_db module

sergey-trukhin opened the pull request:
##### SUMMARY
Added support for _--hex-blob_ option to _mysql_db_ module.

This option is for dump binary columns using hexadecimal notation (for example, 'abc' becomes 0x616263). The affected data types are BINARY, VARBINARY, the BLOB types, and BIT.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
lib/ansible/modules/database/mysql/mysql_db.py

##### ANSIBLE VERSION
2.3.0


##### ADDITIONAL INFORMATION
It could be useful for people who need to make dump of DB which contains binary data because it may cause issues during the restore if you dumped DB with binary data, but without _--hex-blob_ option.

pchauncey reviewed (approved):
LGTM, tested with ansible 2.4.0, mariadb-10.1.21-3.fc25.x86_64, and a blob

pchauncey commented:
shipit

bmalynovytch commented:
shipit

pchauncey commented:
shipit


On Wed, Feb 20, 2019 at 8:16 AM Benjamin MALYNOVYTCH <
notifications@github.com> wrote:

> shipit
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ansible/ansible/pull/23891#issuecomment-465592360>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA3eExOvexKe8oWHb3Wbrl9_DeXApxaZks5vPViwgaJpZM4NFa6t>
> .
>

bmalynovytch commented:
bot_status

Akasurde committed:
mysql_db: Add option hex_blob in db_dump operation

Added support for --hex-blob option to mysql_db module.
This option is for dump binary columns using hexadecimal notation (for example, 'abc' becomes 0x616263).
The affected data types are BINARY, VARBINARY, the BLOB types, and BIT.

Akasurde commented:
@sergey-trukhin Thanks for the contribution.
